pkg/datarepair/checker/checker.go

```diff
@@ -82,7 +82,16 @@ func (c *checker) identifyInjuredSegments(ctx context.Context) (err error) {
 				if err != nil {
 					return Error.New("error unmarshalling pointer %s", err)
 				}
-				pieces := pointer.Remote.RemotePieces
+				remote := pointer.GetRemote()
+				if remote == nil {
+					c.logger.Debug("no remote segment on pointer")
+					continue
+				}
+				pieces := remote.GetRemotePieces()
+				if pieces == nil {
```

Comment:
I wonder if it's okay to have a remote segment with no pieces... Do you think that should be an error?

Reply:
If there are no remote pieces, that just means it's an inline segment, so I think it's fine.

Reply:
ok